crates/data_components/src/spark_connect.rs

```diff
@@ -65,7 +66,9 @@ impl SparkConnect {
     }
 
     pub async fn from_connection(connection: &str) -> Result<Self, Box<dyn Error + Send + Sync>> {
-        let session = Arc::new(SparkSessionBuilder::remote(connection)?.build().await?);
+        let session = SparkSessionBuilder::remote(connection)?.build().await?;
```

Comment:
nit: Reuse of variable name `session` for two different types.

Reply:
According to rust book, variable shadowing is a correct way https://doc.rust-lang.org/stable/book/ch03-01-variables-and-mutability.html#shadowing